src/shared_modules/utils/registryHelper.h

```diff
@@ -199,6 +201,28 @@ namespace Utils
                 return ret;
             }
 
+            bool keyModificationDate(std::string& time) const
+            {
+                auto ret {false};
+                FILETIME lastModificationTime { };
+                auto result
+                {
+                    RegQueryInfoKey(m_registryKey, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, &lastModificationTime)
+                };
+
+                if (ERROR_SUCCESS == result)
+                {
+                    unsigned long long fileTime { lastModificationTime.dwHighDateTime };
+
+                    // Use structure values to build 18-digit LDAP/FILETIME number
+                    fileTime = fileTime << sizeof(lastModificationTime.dwHighDateTime)*8 | lastModificationTime.dwLowDateTime;
```

Comment:
Change approach to the one suggested Microsoft https://docs.microsoft.com/en-us/windows/win32/api/minwinbase/ns-minwinbase-filetime
```
It is not recommended that you add and subtract values from the FILETIME structure to obtain relative times. Instead, you should copy the low- and high-order parts of the file time to a [ULARGE_INTEGER](https://docs.microsoft.com/en-us/windows/win32/api/winnt/ns-winnt-ularge_integer-r1) structure, perform 64-bit arithmetic on the QuadPart member, and copy the LowPart and HighPart members into the FILETIME structure.
```
For example, this is used here 
https://github.com/wazuh/wazuh/blob/3d9e9ac429d439d48506e684eadebd8909727413/src/data_provider/src/sysInfoWin.cpp#L140-L155